### PR TITLE
Windows: Switch UWP build to using winrt generated headers

### PR DESCRIPTION
**Pull request by clarkezone**
This change switches the Windows UWP build from shipped cppwinrt headers to headers generated by the cppwinrt tool.

- Fix support for the latest Windows SDK (10.0.19041.0) by generating winrt SDK headers from SDK metadata rather than using shipped headers which have clang compatibility bugs.  SDK metadata is the source of truth for the WinRT ABI; by using the latest tool rather than shipped headers, most recent compatibility fixes can be picked up on an ongoing basis.

Depends on https://github.com/flutter/buildroot/pull/442

https://github.com/flutter/flutter/issues/70205
https://github.com/flutter/flutter/issues/14967

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [X] I listed at least one issue that this PR fixes in the description above.
- [X] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I signed the [CLA].
- [X] All existing and new tests are passing.
- [X] The reviewer has submitted any presubmit flakes in this PR using the [engine presubmit flakes form] before re-triggering the failure.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/master/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[engine presubmit flakes form]: https://forms.gle/Wc1VyFRYJjQTH6w5A

### CODE REVIEW

File: DEPS [hunk 1]
```diff
@@ -98,7 +98,9 @@ allowed_hosts = [
 ]
 
 deps = {
-  'src': 'https://github.com/flutter/buildroot.git' + '@' + 'd05290fdfb0b8d2aad3c60bf0b4a76f368616452',
+  #remove once https://github.com/flutter/buildroot/pull/442 lands
+  #'src': 'https://github.com/flutter/buildroot.git' + '@' + 'd05290fdfb0b8d2aad3c60bf0b4a76f368616452',
+  'src': 'https://github.com/clarkezone/buildroot.git' + '@' + 'e7d24e56a77178a9fe0a13ee32e2a2ed785628b1',
```
Comment (cbracken) [inline DEPS:103]: Paranoid reminder to fix this before landing :)

**Review by cbracken — APPROVED** [review]
LGTM. Dependent patch https://github.com/flutter/buildroot/pull/442 needs a quick fix then the annotated buildroot roll before landing.

**Comment by chinmaygarde** [comment]
The dependent patch looks good to me after @clarkezone addressed @cbracken s comments on it. I believe this patch is unblocked. Lets land it soon (after fixing the DEPS entry of course 😄 )

**Comment by clarkezone** [comment]
Thx, I'll land both this evening

**Comment by cbracken** [comment]
Kicked build_and_test_linux_unopt_debug. Re-run: https://cirrus-ci.com/task/4666251248140288

**Comment by cbracken** [comment]
Looks like it hung on `ShellTest.SkipAndSubmitFrame`.
```
[ RUN      ] ShellTest.SkipAndSubmitFrame
[FATAL:flutter/testing/test_timeout_listener.cc(45)] Check failed: delay < timeout_. Test ShellTest.SkipAndSubmitFrame did not complete in 30 seconds and is assumed to be hung. Killing the test harness.
```

Filed another flake.

**Pull request merged**